DotNetHowToAutocomplete/DotNetHowToAutocomplete/Controllers/HomeController.cs

```diff
@@ -0,0 +1,159 @@
+
+        private void InitSearch()
+        {
+                string searchServiceName = ConfigurationManager.AppSettings["SearchServiceName"];
+                string apiKey = ConfigurationManager.AppSettings["SearchServiceApiKey"];
+
+                // Create a reference to the NYCJobs index
+                _searchClient = new SearchServiceClient(searchServiceName, new SearchCredentials(apiKey));
+                _indexClient = _searchClient.Indexes.GetClient(IndexName);
+        }
+
+        public ActionResult Index()
+        {
+            return View();
+        }
+
+        public ActionResult IndexJavaScript()
+        {
+            return View();
+        }
+
+        public ActionResult About()
+        {
+            ViewBag.Message = "Your application description page.";
+
+            return View();
+        }
+
+        public ActionResult Contact()
+        {
+            ViewBag.Message = "Your contact page.";
+
+            return View();
+        }
+        public ActionResult Suggest(bool highlights, bool fuzzy, string term)
+        {
+            InitSearch();
+
+            // Call suggest API and return results
+            SuggestParameters sp = new SuggestParameters()
+            {
+                UseFuzzyMatching = fuzzy,
+                Top = 5
+            };
+
+            if (highlights)
+            {
+                sp.HighlightPreTag = "<b>";
+                sp.HighlightPostTag = "</b>";
+            }
+
+            DocumentSuggestResult resp = _indexClient.Documents.Suggest(term, "sg", sp);
+
+            // Convert the suggest query results to a list that can be displayed in the client.
+            List<string> suggestions = resp.Results.Select(x => x.Text).ToList();
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = suggestions
+            };
+        }
+
+        public ActionResult AutoComplete(string term)
+        {
+            InitSearch();
+            //Call autocomplete API and return results
+            AutocompleteParameters sp = new AutocompleteParameters()
+            {
+                AutocompleteMode = AutocompleteMode.OneTermWithContext,
+                UseFuzzyMatching = false,
+                Top = 5,
+                MinimumCoverage = 80
+            };
+            AutocompleteResult resp = _indexClient.Documents.Autocomplete(term, "sg", sp);
+
+            // Conver the Suggest results to a list that can be displayed in the client.
+            List<string> autocomplete = resp.Results.Select(x => x.Text).ToList();
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = autocomplete
+            };
+        }
+
+        public ActionResult Both(string term)
+        {
+            InitSearch();
+
+            AutocompleteParameters sp1 = new AutocompleteParameters()
```

Comment:
How about `ap` and `autocompleteResult` instead of `sp1` and `resp1`, and `sp` and `suggestResult` instead of `sp2` and `resp2`?